Add a dedicated client to communicate with the Proxy SSH servers

rosstimothy opened the pull request:
A new `api/client/proxy/Client` has been added to interact with
the SSH and gRPC servers that the Proxy serves on its SSH port.
The client will first try connecting to the gRPC server and if
that fails it will fall back to the SSH server - this ensures backwards
compatibility. 

Much of the SSH functionality mimics the existing behavior of the
`ProxyClient` in `lib/client`. This is the first part of phasing
out that client in favor of the new client. There will be a follow
up PR that migrates `lib/client` to make use of the new client introduced 
here.

Part of https://github.com/gravitational/teleport/issues/19812


**Note**:  The first commit moves `api/client/proxy/proxy.go` to `api/utils.proxy.go` in order to avoid circular dependencies.

gzdunek reviewed (approved):
Approved, but I'm not familiar with this part of the codebase, so you may want to request a review from someone else from group 2.

timothyb89 reviewed (approved):
This looks reasonable. I'm excited to see this in `tsh ssh`!

rosstimothy committed:
Add a dedicated client to communicate with the Proxy SSH server

A new `api/client/proxy/Client` has been added to interact with
the SSH and gRPC servers that the Proxy serves on its SSH port.
The client will first try connecting to the gRPC server and if
that fails it will fall back to the SSH server.

Much of the SSH functionality mimics the existing behavior of the
`ProxyClient` in `lib/client`. This is the first part of phasing
out that client in favor of the new client. There will be a follow
up PR that migrates `lib/client` to make use of the new client instead.

Part of #19812
